File: solutions/fortran/collatz-conjecture/1/collatz_conjecture.f90
module collatz_conjecture
  implicit none
contains

  integer function steps(i)
    integer :: i, j
    !check for neg or zero
    if (i<1) then
      steps = -1
    else
      j=i
      steps=0
      do while (j>1)
        if (mod(j,2)==0) then
          j=j/2
        else
          j=3*j+1
        end if
        steps=steps+1
      end do
    end if
  end function

end module

File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    int steps=0;
    if (start < 1) steps=ERROR_VALUE;
    else {
        while (start > 1) {
            if (start % 2 == 0) start=start/2;
            else start=3*start+1;
            steps++;
        }
    }
    return steps;
}
